### DISCUSSION TITLE
Currency format without rounding

### DISCUSSION THREAD

**Discussion by TheRoSS** [grafana/grafana#38355] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
We need currency formats like $1 000 000

**Why is this needed**:
At present the currency formats rounds the numbers. So $1 254 998 becomes $1M. This is not acceptable for financial reports. We want to use currency signs (we work with different currencies) but with strict values (without rounding) and spaces between digit categories. The only way to do it now is to format sql select part as string. This is very uncomfortable

PS. Your formats are hard-coded. So there is no way to change the formatting or add new one. The best solution as I see would be to add such a feature. There should be some default formattings and some rules to add new ones (by plugin for example)

**Comment by TheRoSS** [comment]
If I would solve this problem I would create an angular service to deal with all formats and make it accessable to plugins via injection. Then I would create a panel plugin and edit formatters via this service. Installing this plugin will unlock my format changes to dashboard

But as I understand you try to avoid angular using as much as possible. And I don't know how to implement such feature in react. Otherwise I could try to write a patch

**Comment by ryantxu** [comment]
In master (nightly), can you try entering the unit "prefix:$" and see if that does what you want.

Re angular -- yes we are getting rid of as much angular as possible, any new contribution will be ts or react

**Comment by TheRoSS** [comment]
Where can I see this unit? I look at plugin table (grafana 6.6.0, just updated master branch) and don't see such an option

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). I believe this has been implemented. If not, feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
